src/objective.jl

```diff
@@ -160,3 +160,34 @@ function objective_function(model::Model,
                    MOI.ObjectiveFunction{MOIFunType}())::MOIFunType
     return jump_function(model, func)
 end
+
+"""
+    set_objective_coefficient(model::Model, term, coeff)
+
+Updates the objective function to change the coefficient of the given term.
+If the objective does not have the right type, an error will be thrown:
+changing a variable coefficient for a constant objective, changing the
+coefficient of a quadratic term where there is only a linear objective.
+"""
+function set_objective_coefficient end
+
+function set_objective_coefficient(model::Model, variable::VariableRef, coeff)
+    if model.nlp_data !== nothing && model.nlp_data.nlobj !== nothing
+        error("A nonlinear objective is already set in the model")
+    end
+
+    obj_fct_type = moi_function_type(objective_function_type(model))
+    if obj_fct_type == MOI.SingleVariable
+        # Promote the objective function to be an affine expression.
+        set_objective_function(model, coeff * variable)
```

Comment:
What if the single variable is not `variable`?
```Julia
model = Model()
@variable(model, x)
@variable(model, y)
@objective(model, Min, x)
set_objective_coefficient(model, y, 2.0)
@test JuMP.isequal_canonical(JuMP.objective_function(m), x + 2.0 * y)
```

Reply:
I didn't think of that case, thanks!